Bottom: 799817db26572836f175d32bdd60ba07cd85538c
Top:    c9c47c48f54de5e9e51d0270216f3c5b3ee77700
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-26 23:47:22 +0100

Create new function GetLayerGroupNumberBySide()

Makes the intention of various code clearer.


---

diff --git a/src/action.c b/src/action.c
index e496820..9c3acd2 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1386,7 +1386,7 @@ NotifyMode (void)
 		int flag = CLEARLINEFLAG;
 
 		if (GetLayerGroupNumberByNumber (INDEXOFCURRENT) ==
-		    GetLayerGroupNumberByNumber (solder_silk_layer))
+		    GetLayerGroupNumberBySide (SOLDER_LAYER))
 		  flag |= ONSOLDERFLAG;
 		if ((text = CreateNewText (CURRENT, &PCB->Font, Note.X,
 					   Note.Y, 0, Settings.TextScale,
diff --git a/src/misc.c b/src/misc.c
index 064c3e4..95806bb 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1435,6 +1435,19 @@ GetLayerGroupNumberByNumber (Cardinal Layer)
 }
 
 /* ---------------------------------------------------------------------------
+ * returns the layergroup number for the passed side (TOP_LAYER or BOTTOM_LAYER)
+ */
+int
+GetLayerGroupNumberBySide (int side)
+{
+  /* Find the relavant board side layer group by determining the
+   * layer group associated with the relevant side's silk-screen
+   */
+  return GetLayerGroupNumberByNumber(
+      side == COMPONENT_LAYER ? component_silk_layer : solder_silk_layer);
+}
+
+/* ---------------------------------------------------------------------------
  * returns a pointer to an objects bounding box;
  * data is valid until the routine is called again
  */
diff --git a/src/misc.h b/src/misc.h
index 9ce0905..25db4dc 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -74,6 +74,7 @@ void SaveOutputWindow (void);
 int GetLayerNumber (DataType *, LayerType *);
 int GetLayerGroupNumberByPointer (LayerType *);
 int GetLayerGroupNumberByNumber (Cardinal);
+int GetLayerGroupNumberBySide (int);
 int GetGroupOfLayer (int);
 int ChangeGroupVisibility (int, bool, bool);
 void LayerStringToLayerStack (char *);
